src/main/java/com/cloudbees/jenkins/plugins/awscredentials/AWSCredentialsImpl.java

```diff
@@ -126,6 +126,10 @@ public boolean requiresToken() {
     }
 
     public AWSCredentials getCredentials() {
+        return getCredentials(this.getStsTokenDuration());
+    }
+
+    public AWSCredentials getCredentials(Integer stsTokenDuration) {
```

Comment:
Probably `int` not `Integer`, or can this be null?

Reply:
I don't think it should generally be `null`. Probably better to prevent this.

I had used `Integer` rather than `int`, as it seems to be used pretty consistently across the project. For the moment, I've added the `@NonNull` annotation on it. Do you think this would be sufficient, or do you think I should still change it to `int`?

Reply:
Should be `int` unless you have a clear use case for a null value.

Reply:
Okay. I'll change it